TypeCobol/Compiler/Parser/DiagnosticUtils.cs

```diff
@@ -28,7 +29,13 @@ internal static class DiagnosticUtils
             var parserDiag = new ParserDiagnostic(message, data.DataName.NameLiteral.Token, rulestack, code);
             e.Diagnostics.Add(parserDiag);
         }
-
+	    internal static void AddErrorWithNoRuleStack(CodeElement e, string message, Antlr4.Runtime.RuleContext context, MessageCode code = MessageCode.SyntaxErrorInParser)
```

Comment:
I'm not sure a new method is required here because several methods AddError accept a null value for the ruleStack parameter.

You can update the method
```csharp
internal static void AddError(CodeElement e, string message, Antlr4.Runtime.RuleContext context, MessageCode code = MessageCode.SyntaxErrorInParser)
```
with an optional `includeRuleStack` boolean parameter (with default value to true).